test/sentry/transport_test.exs

```diff
@@ -190,11 +190,17 @@ defmodule Sentry.TransportTest do
         Plug.Conn.resp(conn, 200, ~s<invalid JSON>)
       end)
 
-      assert {:request_failure, %Jason.DecodeError{}} =
+      assert {:request_failure, error} =
                error(fn ->
                  Transport.encode_and_post_envelope(envelope, HackneyClient, _retries = [0])
                end)
 
+      if Version.match?(System.version(), "~> 1.18") do
+        assert error.__struct__ == JSON.DecodeError
+      else
+        assert error.__struct__ == Jason.DecodeError
+      end
```

Comment:
Given that we're introducing a shim under `Sentry.JSON` shouldn't we also raise `Sentry.JSON.DecodeError`?

Reply:
We're not introducing `Sentry.JSON` for users, it's just for us during this transition time.